From: Nick Rosbrook <enr0n@ubuntu.com>
Date: Thu, 18 Apr 2024 12:01:42 -0400
Subject: copy: ignore -EOPNOTSUPP from copy_file_range()

Origin: upstream, https://github.com/systemd/systemd/commit/c0bc1e897178da521e74acb270843805f5906adf
Bug-Ubuntu: https://launchpad.net/bugs/2058179

According to copy_file_range (2), errno will be set to EOPNOTSUPP when
the file system does not support copy_file_range(). Since there is
already fallback logic in place here for other kinds of errors, add
-EOPNOTSUPP to the list of ignored errors.
---
 src/shared/copy.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/shared/copy.c b/src/shared/copy.c
index c0e30cd..bc8643e 100644
--- a/src/shared/copy.c
+++ b/src/shared/copy.c
@@ -316,7 +316,7 @@ int copy_bytes_full(
                 if (try_cfr) {
                         n = try_copy_file_range(fdf, NULL, fdt, NULL, m, 0u);
                         if (n < 0) {
-                                if (!IN_SET(n, -EINVAL, -ENOSYS, -EXDEV, -EBADF))
+                                if (!IN_SET(n, -EINVAL, -ENOSYS, -EXDEV, -EBADF, -EOPNOTSUPP))
                                         return n;
 
                                 try_cfr = false;
